### PR TITLE
Simplify VirtNetwork._lock() and MachineSSHConnection.execute()

### PR DESCRIPTION
**Pull request by martinpitt**
Needs these PRs landed first:
 - #3706
 - #3707 
 - https://github.com/osbuild/cockpit-composer/pull/1477
 - https://github.com/cockpit-project/cockpit/pull/17068

### CODE REVIEW

File: machine/machine_core/ssh_connection.py [hunk 1]
```diff
@@ -306,70 +301,26 @@ def execute(self, command=None, script=None, input=None, environment={},
...
-
-        if proc.returncode != 0 and check:
-            raise subprocess.CalledProcessError(proc.returncode, command, output=output)
-        return output
+            res = subprocess.run(command_line,
+                                 stdin=None if input is not None else subprocess.DEVNULL,
+                                 input=input,
+                                 stdout=stdout or subprocess.PIPE,
+                                 check=check,
+                                 text=True)
```
Comment (martinpitt) [inline machine/machine_core/ssh_connection.py:321]: Using `text=true` would have been nice, but we have [some tests](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3709-20220811-142551-14c3d745-rhel-8-7-cockpit-project-cockpit/log.html) that produce binary garbage. So I'm going to bring back the original `decode('utf-8', 'replace')`.

Comment (martinpitt) [reply machine/machine_core/ssh_connection.py]: Oh, `subprocess.run()` has an `errors=` kwarg, nice! (they really thought about everything..)

Comment (martinpitt) [reply machine/machine_core/ssh_connection.py]: Bah -- [this test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3709-20220812-053048-dd254213-debian-stable-cockpit-project-cockpit/log.html#19-2) is too strict and expects `\r\n`, which `text=True` eats up. So, back to manual .decode() after all..

**Comment by martinpitt** [comment]
[this openstack composer failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3709-20220812-064143-5c2df293-fedora-36-osbuild-cockpit-composer/log.html#20-2) is unrelated, and smells more like changing/flaking infra. Retrying.

**Comment by martinpitt** [comment]
@jkozol : seems the [openstack test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-3709-20220812-083315-5c2df293-fedora-36-osbuild-cockpit-composer/log.html) is broken again.. :cry:

File: machine/machine_core/ssh_connection.py [hunk 2]
```diff
@@ -306,70 +301,25 @@ def execute(self, command=None, script=None, input=None, environment={},
...
-            if getattr(command, "strip", None):  # Is this a string?
-                cmd += [command]
-                if not quiet:
-                    self.message("+", command)
-            else:
-                # use shlex.join() once Python 3.8 is available everywhere
-                cmd.append(' '.join(shlex.quote(arg) for arg in command))
-                if not quiet:
-                    self.message("+", *command)
+        if getattr(command, "strip", None):  # Is this a string?
```
Comment (allisonkarlitskaya) [inline machine/machine_core/ssh_connection.py:304]: I could have sworn we changed this to `isinstance()`...

**Comment by jkozol** [comment]
@martinpitt the openstack test can be ignored for now. As soon as this PR: https://github.com/osbuild/cockpit-composer/pull/1443 is reviewed and I merge to main to the tests will be fixed. We override rhel's default cdn repos with osbuild's self hosted snapshot repos. The gpg key for 9.1 was wrong causing the 9.1 image to fail to build. This is fixed as part of https://github.com/osbuild/cockpit-composer/pull/1443.

**Review by allisonkarlitskaya — APPROVED** [review]
This is a nice cleanup.

I had some thoughts:

 - did you consider using `communicate()` instead?
 - did you consider always providing an `input=` kwarg, possibly as an empty string, and always giving `stdin=None`?

But in both cases, I think I actually prefer your approach.

File: machine/machine_core/ssh_connection.py (hunk 2 above)
Comment (martinpitt) [reply machine/machine_core/ssh_connection.py:304]: No, I backed out that commit from PR #3706 as you weren't happy about the aliasing, and I wasn't happy with any other solution.

**Comment by martinpitt** [comment]
@allisonkarlitskaya: I didn't consider these. Feel free to change them around, of course.

**Pull request merged**